segmentation_models_pytorch/encoders/sam.py

```diff
@@ -0,0 +1,98 @@
+import math
+import warnings
+from typing import Mapping, Any
+
+import torch
+from segment_anything.modeling import ImageEncoderViT
+
+from segmentation_models_pytorch.encoders._base import EncoderMixin
+
+
+class SamVitEncoder(EncoderMixin, ImageEncoderViT):
+    def __init__(self, name: str, **kwargs):
+        patch_size = kwargs.get("patch_size", 16)
+        n_skips = kwargs.pop("num_hidden_skips", int(self._get_scale_factor(patch_size)))
```

Comment:
Do we actually need this num_hidden_skips? Can it be estimated based on depth?

Reply:
I'm guessing all these three issues can be resolved by creating an extra param `vit_depth` which will be unrelated to `encoder_depth`, and handle them separately. This should help to remove these code chunks but creates a slight ambiguity (which I can explain in the docstring). What do you think?

Reply:
Yes, I like that idea, I think this would be better to have this logic in one place, rather than split into different places

Reply:
all fixed now